internal/kafka/internal/workers/cluster_mgrs/clusters_mgr.go

```diff
@@ -854,6 +860,33 @@ func (c *ClusterManager) buildObservabilityCloudwatchLoggingCredentialsSecret()
 	}
 }
 
+func (c *ClusterManager) buildObservabilityEnterpriseCloudwatchLoggingCredentialsSecret(orgID string, clusterID string) *k8sCoreV1.Secret {
+	cloudwatchCredentialSecret := &k8sCoreV1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: k8sCoreV1.SchemeGroupVersion.String(),
+			Kind:       "Secret",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      c.ObservabilityConfiguration.ObservabilityCloudWatchLoggingConfig.K8sCredentialsSecretName,
+			Namespace: c.ObservabilityConfiguration.ObservabilityCloudWatchLoggingConfig.K8sCredentialsSecretNamespace,
+		},
+		Type: k8sCoreV1.SecretTypeOpaque,
+	}
+
+	credentials := c.ObservabilityConfiguration.ObservabilityCloudWatchLoggingConfig.GetEnterpriseCredentials(orgID)
+	if credentials.AccessKey != "" && credentials.SecretAccessKey != "" {
```

Comment:
Easier/simpler alternatives to this are:
* Change the GetEnterpriseCredentials signature to return `*ObservabilityCloudwatchLoggingConfigCredentials` instead of `ObservabilityCloudwatchLoggingConfigCredentials`, where `nil` means that it hasn't been found

* Change the GetEnterpriseCredentials signature to return `ObservabilityCloudwatchLoggingConfigCredentials, bool` instead of `ObservabilityCloudwatchLoggingConfigCredentials`, where the bool indicates whether the credentials were found. This is similar to how lookups are done in Go map data types.

Both approaches are fine to me.

Reply:
:+1: